src/bits.rs

```diff
@@ -13,19 +13,40 @@
 
 
 /// `bits!( parser ) => ( &[u8], (&[u8], usize) -> IResult<(&[u8], usize), T> ) -> IResult<&[u8], T>`
-/// transforms its byte slice input into a bit stream for the underlying parsers
+/// Transforms its byte slice input into a bit stream for the underlying parsers.
 ///
 /// ```
 /// # #[macro_use] extern crate nom;
 /// # use nom::IResult::Done;
 /// # fn main() {
 ///  named!( take_3_bits<u8>, bits!( take_bits!( u8, 3 ) ) );
 ///
-///  let input = vec![0b10101010, 0b11110000, 0b00110011];
-///  let sl    = &input[..];
+///  let bytes = vec![0b1010_1010, 0b1111_0000, 0b0011_0011];
+///  let slice = &bytes[..];
+///
+///  assert_eq!( take_3_bits( slice ), Done(&slice[1..], 5) );
+/// # }
+/// ```
+///
+/// The consumed bits counter is local to a `bits!` macro invocation, so the remaining result
+/// is not `(&[u8], usize)` anymore. Second element is bit index in the stream - essential to parse
+/// all the bits or else they will be dropped.
 ///
-///  assert_eq!(take_3_bits( sl ), Done(&sl[1..], 5) );
+/// ```
+/// # #[macro_use] extern crate nom;
+/// # use nom::IResult::Done;
+/// # fn main() {
+///  named!( take_z_word<(u8, u16)>,
```

Comment:
The name `take_z_word` is from a project I was working on when I asked the question that led to this documentation change. (I am writing a parser for files from the game Zork.) But this name will not mean anything to most people. Is there a more generic name we could use?